### DISCUSSION TITLE
Order of State Updates

### DISCUSSION THREAD

**Discussion by cort1c** [colyseus/colyseus#788] [Q&A] [answered]
I was looking through the documentation but didn't see anything definitive about this. Basically, my question is, are client-side sync callbacks executed in the same order in which the state is updated on the server?

For example if I do something like the following on the server...

```
playerState.Health = 0;
playerState.Dead = true;
```

...can I assume OnHealthChange will be called **before** OnDeadChanged on the client?

Also, a related question, is **all** state synchronized for the current patch **before** any callbacks are executed?

Thanks!

**Comment by endel** [comment, marked as answer]
Hi @cort1c 👋  You are right on both assumptions!

- All state is decoded & applied in the client-side **before** triggering any callbacks
- The order of callbacks triggered in the client-side are the same as they were mutated in the server-side

Cheers

**Comment by cort1c** [comment]
Thank you!
